hwmv2: Port STM32F4X SoCs & boards

ajarmouni-st opened the pull request:
Port STM32F4X SoCs & boards to Hardware Model V2.

ajarmouni-st committed:
soc: v2: stm32: Migrate STM32F4 series

Port STM32F4 series to HW model V2

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert 96b_aerocore2 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert 96b_argonkey to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert 96b_carbon to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert 96b_neonkey to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert 96b_stm32_sensor_mez to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert adafruit_feather_stm32f405 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert adi_sdp_k1 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert az3166_iotdevkit to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert black_f407ve to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert black_f407zg_pro to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert blackpill_f401cc to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert blackpill_f401ce to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert blackpill_f411ce to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert google_dragonclaw to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert mikroe_clicker_2 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert mikroe_mini_m4_for_stm32 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert nucleo_f401re to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert nucleo_f410rb to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert nucleo_f411re to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert nucleo_f412zg to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert nucleo_f413zh to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert nucleo_f429zi to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert nucleo_f446re to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert nucleo_f446ze to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert olimex_stm32_e407 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert olimex_stm32_h405 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert olimex_stm32_h407 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert olimex_stm32_p405 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert segger_trb_stm32f407 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert steval_fcu001v1 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert stm32f401_mini to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert stm32f411e_disco to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert stm32f412g_disco to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert stm32f429i_disc1 to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert stm32f469i_disco to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

ajarmouni-st committed:
boards: Convert stm32f4_disco to HWM v2

Port the board to HWMv2.

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>

erwango commented on soc/st/stm32/stm32f4/Kconfig.defconfig.stm32f407xx:
```diff
@@ -5,19 +5,13 @@
 
 if SOC_STM32F407XE
```
if SOC_STM32F407XE || SOC_STM32F407XG

erwango commented on soc/st/stm32/stm32f4/Kconfig.defconfig.stm32f412xx:
```diff
@@ -5,29 +5,20 @@
 
 if SOC_STM32F412CX
```
if SOC_STM32F412CX || SOC_STM32F412VX || SOC_STM32F412ZX

erwango commented on soc/st/stm32/stm32f4/Kconfig.soc:
```diff
@@ -0,0 +1,113 @@
+config SOC
+	default "stm32f401xc" if SOC_STM32F401XC
+	default "stm32f401xe" if SOC_STM32F401XE
+	default "stm32f405xx" if SOC_STM32F405XG
```
Let's take the opportunity to fix it as SOC_STM32F405XX.
To be done as a follow up commit to make it visible

erwango commented on soc/st/stm32/stm32f4/Kconfig.soc:
```diff
@@ -0,0 +1,113 @@
+	default "stm32f401xc" if SOC_STM32F401XC
+	default "stm32f401xe" if SOC_STM32F401XE
+	default "stm32f405xx" if SOC_STM32F405XG
+	default "stm32f407xx" if SOC_STM32F407XE
```
Same here for SOC_STM32F407XE. Could be same commit as SOC_STM32F405XX

ajarmouni-st replied:
SOC_STM32F407XE & SOC_STM32F407XG are used in here https://github.com/zephyrproject-rtos/zephyr/blob/129b73ce266e1eabc518333b38d2971c207685fa/Kconfig.zephyr#L666-L667

erwango replied:
These should be changed as well. 
(And actually, I think this property setting should be moved to soc, since there is no way one can remember that it has to be set, but that's for a latter PR)

erwango commented on soc/st/stm32/stm32f4/Kconfig.soc:
```diff
@@ -0,0 +1,113 @@
+	default "stm32f401xe" if SOC_STM32F401XE
+	default "stm32f405xx" if SOC_STM32F405XG
+	default "stm32f407xx" if SOC_STM32F407XE
+	default "stm32f407xx" if SOC_STM32F407XG
```
Ditto

erwango reviewed (changes requested):
Otherwise LGTM

ajarmouni-st committed:
soc: st: stm32: stm32f4: change SOC_STM32F405XG to SOC_STM32F405XX

Change SOC_STM32F405XG flag to SOC_STM32F405XX since it refers to SoC
stm32f405xx in kconfig.soc

Signed-off-by: Abderrahmane Jarmouni <abderrahmane.jarmouni-ext@st.com>
